src/main/java/org/broadinstitute/hellbender/tools/spark/sv/evidence/FindBadGenomicKmersSpark.java

```diff
@@ -77,11 +84,63 @@ protected void runTool( final JavaSparkContext ctx ) {
                                              final ReferenceMultiSource ref,
                                              final SAMSequenceDictionary readsDict ) {
         // Generate reference sequence RDD.
-        final JavaRDD<byte[]> refRDD = SVReferenceUtils.getRefRDD(ctx, kSize, ref, readsDict,
-                                                        SVReferenceUtils.REF_RECORD_LEN, SVReferenceUtils.REF_RECORDS_PER_PARTITION);
+        final SAMSequenceDictionary dict = ref.getReferenceSequenceDictionary(readsDict);
+        if ( dict == null ) throw new GATKException("No reference dictionary available");
+        final JavaRDD<byte[]> refRDD = SVReferenceUtils.getReferenceBasesRDD(ctx, kSize, ref, dict,
+                                                        REF_RECORD_LEN, REF_RECORDS_PER_PARTITION);
 
         // Find the high copy number kmers
-        return SVReferenceUtils.processRefRDD(kSize, maxDUSTScore, MAX_KMER_FREQ, refRDD);
+        return collectUbiquitousKmersInReference(kSize, maxDUSTScore, MAX_KMER_FREQ, refRDD);
+    }
+
+    /**
+     * Do a map/reduce on an RDD of genomic sequences:
+     * Kmerize, mapping to a pair <kmer,1>, reduce by summing values by key, filter out <kmer,N> where
+     * N <= MAX_KMER_FREQ, and collect the high frequency kmers back in the driver.
+     */
+    public static List<SVKmer> collectUbiquitousKmersInReference(final int kSize,
+                                                                 final int maxDUSTScore,
+                                                                 final int maxKmerFreq,
+                                                                 final JavaRDD<byte[]> refRDD) {
+        Utils.nonNull(refRDD, "reference bases RDD is null");
+        Utils.validateArg(kSize > 0, "provided kmer size is non positive");
+        Utils.validateArg(maxDUSTScore > 0, "provided DUST filter score is non positive");
+        Utils.validateArg(maxKmerFreq > 0, "provided kmer frequency is non positive");
+
+        final int nPartitions = refRDD.getNumPartitions();
+        final int hashSize = 2*REF_RECORDS_PER_PARTITION;
+        return refRDD
+                .mapPartitions(seqItr -> {
+                    final HopscotchMap<SVKmer, Integer, KmerAndCount> kmerCounts = new HopscotchMap<>(hashSize);
+                    while ( seqItr.hasNext() ) {
+                        final byte[] seq = seqItr.next();
```

Comment:
What about:
```
Utils.stream(seqItr)
       .flatMap(seq -> SVDUSTFilteredKmerizer.canonicalStream(seq, kSize, maxDUSTScore, new SVKmerLong())
       .forEach(kmer -> {...});
```

If one defines a ```Hopscotch.bumpAll(Hopscotch<K, V, T> other)``` that would add the counts from the other hop-scotch then one could use the ```Stream.reduce(Unit, Add, Combine)``` method instead than the ```forEach```.

Reply:
not going to change implementations in this PR, since this method is not being refactored into Utils any more. Trying to limit what this PR does.